Sources/MySQL/Database/MySQLDatabaseConfig.swift

```diff
@@ -36,4 +36,38 @@ public struct MySQLDatabaseConfig {
         self.capabilities = capabilities
         self.characterSet = characterSet
     }
+
+    /// Creates a `MySQLDatabaseConfig` frome a connection string.
+    public init(url urlString: String, capabilities: MySQLCapabilities = .default, characterSet: MySQLCharacterSet = .utf8_general_ci) throws {
+        guard
+            let url = URL(string: urlString),
+            let hostname = url.host,
+            let port = url.port,
+            let username = url.user,
+            url.path.count > 0
+        else {
+            throw MySQLError(
+                identifier: "Bad Connection String",
+                reason: "Host could not be parsed",
+                possibleCauses: ["Foundation URL is unable to parse the provided connection string"],
+                suggestedFixes: ["Check the connection string being passed"],
+                source: .capture()
+            )
+        }
+
+        self.hostname = hostname
+        self.port = port
+        self.username = username
+        self.password = url.password
+        self.capabilities = capabilities
+        self.characterSet = characterSet
+
+        let database = url.path
+
+        if database.hasPrefix("/") {
```

Comment:
You should use this instead: https://api.vapor.codes/database-kit/latest/DatabaseKit/Extensions/URL.html#/s:11DatabaseKit12databaseNameSSSgv

Reply:
Ah, I didn’t know about that method, fixed!

Reply:
… although now even the test on MySQL 5.5 is failing, but I don’t think it’s because of this change.